src/s1reader/s1_reader.py

```diff
@@ -682,17 +785,48 @@ def burst_from_xml(annotation_path: str, orbit_path: str, tiff_path: str,
     # calculate the range at mid swath (mid of SM swath, mid of IW2 or mid of EW3)
     with open_method(iw2_annotation_path, 'r') as iw2_f:
         iw2_tree = ET.parse(iw2_f)
-        iw2_slant_range_time = float(iw2_tree.find('imageAnnotation/imageInformation/slantRangeTime').text)
+        iw2_slant_range_time =\
+            float(iw2_tree.find('imageAnnotation/imageInformation/slantRangeTime').text)
         iw2_n_samples = int(iw2_tree.find('swathTiming/samplesPerBurst').text)
         iw2_starting_range = iw2_slant_range_time * isce3.core.speed_of_light / 2
         iw2_mid_range = iw2_starting_range + 0.5 * iw2_n_samples * range_pxl_spacing
 
     # find orbit state vectors in 'Data_Block/List_of_OSVs'
     if orbit_path:
         orbit_tree = ET.parse(orbit_path)
-        osv_list = orbit_tree.find('Data_Block/List_of_OSVs')
+        orbit_state_vector_list = orbit_tree.find('Data_Block/List_of_OSVs')
+
+        # Calculate ascending node crossing time from orbit;
+        # compare with the info from annotation
+        try:
+            ascending_node_time_orbit = get_ascending_node_time_orbit(
+                                            orbit_state_vector_list,
+                                            first_line_utc_time,
+                                            ascending_node_time_annotation)
+            ascending_node_time = ascending_node_time_orbit
+
+        except ValueError:
+            warnings.warn('Cannot estimate ascending node crossing time from Orbit. '
+                          'Using the ascending node time in annotation.')
+            ascending_node_time_orbit = None
+            ascending_node_time = ascending_node_time_annotation
+
+        # Calculate the difference in the two ANX times, and give
+        # warning message when the difference is noticeable.
+        if not ascending_node_time_orbit is None:
```

Comment:
```suggestion
        if ascending_node_time_orbit is not None:
```

https://peps.python.org/pep-0008/#programming-recommendations recommends to use `is not None`